rllib/algorithms/dreamerv3/tf/dreamerv3_tf_learner.py

```diff
@@ -767,21 +859,32 @@ def _compute_scaled_value_targets(
         Per_R_5 = tfp.stats.percentile(value_targets_H_B, 5)
         Per_R_95 = tfp.stats.percentile(value_targets_H_B, 95)
 
-        # Update EMAs stored in actor network.
-        # Initial values: Just set.
-        if tf.math.is_nan(actor.ema_value_target_pct5):
-            actor.ema_value_target_pct5.assign(Per_R_5)
-            actor.ema_value_target_pct95.assign(Per_R_95)
-        # Later update (something already stored in EMA variable): Update EMA.
-        else:
-            actor.ema_value_target_pct5.assign(
+        # Update EMA values for 5 and 95 percentile, stored as tf variables under actor
+        # network.
+        # 5 percentile
+        new_val_pct5 = tf.where(
```

Comment:
This if-block seemed to cause trouble (b/c of autograph?). Either way, I replaced with a standard `tf.where` clause.